Convert Chromband.Rnw to Chromband.Rmd

kemicky opened the pull request:
@Bioconductor/sweave2rmd: please may you review Chromband sweave2rmd ?, thank you
[ChromBand.pdf](https://github.com/Bioconductor/Category/files/11057856/ChromBand.pdf)

[ChromBand.html](https://github.com/kemicky/Category/blob/file-Rmd/vignettes/ChromBand%20Html%20%26%20Pdf%20file/Using%20Chromosome%20Bands%20as%20Categories.webarchive)

jwokaty committed:
first commit

kemicky committed:
file clean-up

kemicky committed:
Remove ChromBand.Rnw

jwokaty commented:
Hi @kemicky, good job making the PR.

Your PR should include
* addition of the new .Rmd file
* removal of the .Rnw file
* edits to the DESCRIPTION file

No other files should be added to or altered in this PR. Usually files like this get included by doing `git add .`. It's good to get in the habit of explicitly writing out the files you want to include so that you don't make unnecessary additions. Can you please take care of these extra files?

kemicky commented:
> Hi @kemicky, good job making the PR.
> 
> Your PR should include
> 
> * addition of the new .Rmd file
> * removal of the .Rnw file
> * edits to the DESCRIPTION file
> 
> No other files should be added to or altered in this PR. Usually files like this get included by doing `git add .`. It's good to get in the habit of explicitly writing out the files you want to include so that you don't make unnecessary additions. Can you please take care of these extra files?

Good day everyone, thank you for your review, I did the edits to the description file, removed the Chromband.Rnw file , added the .Rmd. file. and as instructed I will remove the extra files.
 But please to clarify the new ChromBand.Rmd file , I created the table using the Chromband.xlsx file and as well added the image using the image in the figure files so if I remove them might alter the table and images, please any advice on that too. Thank you very much.

jwokaty commented:
I looked at the existing files in the devel branch and no images exist there, so we should be able to generate them by compiling the .Rmd. Anything that we can regenerate, we don't commit. If they are not being regenerated, we will have to fix this during the review.

kemicky commented:
The Images and the table are included in the **original .rnw  file as figure and tb**  but after they were converted to .Rmd they were not formatted correctly, So I only just fixed it to look as the pdf as instructed in the contribute article.  
May you please see, **here is a screenshot of the original ChromBand.Rnw**
![image of the original ChromBand Rnw file](https://user-images.githubusercontent.com/57273086/227655858-a50be39a-35c9-4bf5-aaa3-63bbb731823e.png)

jwokaty commented on vignettes/ChromBand.Rmd:
````diff
@@ -0,0 +1,578 @@
+
+Figure [1](#fig:chr12ideogram){reference-type="ref" reference="fig:chr12ideogram"} shows an ideogram of human chromosome 12, with the band 12q21 shaded. As shown in the figure, 12q21 can be divided into more granular levels 12q21.1, 12q21.2, and 12q21.3. 12q21.3 can itself be divided at an even finer level of resolution into 12q21.31, 12q21.32, and 12q21.33. Moving towards less granular bands, 12 ...
+
+```{r,echo=FALSE, eval=FALSE}
+if (!requireNamespace("BiocManager", quietly = TRUE))
+    install.packages("BiocManager")
+BiocManager::install("Category")
+```
+
````
Remove lines 30-35.  They are not in the existing PDF.

jwokaty commented on vignettes/ChromBand.Rmd:
````diff
@@ -0,0 +1,578 @@
+BiocManager::install("Category")
+```
+
+::: Schunk
+::: Sinput
+\> library("Category")\
+\> library("ALL")\
+\> library("hgu95av2.db")\
+\> library("annotate")\
+\> library("genefilter")\
+\> ##library("SNPchip")\
+\> library("karyoploteR")\
+\> library("geneplotter")\
+\> library("limma")\
+\> library("lattice")\
+\> library("graph")\
+:::
+:::
````
Convert lines 36-50 into a code block. The code for the figures depends on this code.

It should resemble the following

    ```{r setup}
    library("Category")
    library("ALL")
    ```
    
Also, later in the code you have another code block named as `setup`, which will error when you try to knit. This code block should be named `setup` like the original document. I recommend removing the other code block as it sets `echo = TRUE` for all the code blocks; however, that is the default setting.

jwokaty commented on vignettes/ChromBand.Rmd:
````diff
@@ -0,0 +1,578 @@
+
+For illustration, we use a microarray dataset [@Chiaretti2005] from a clinical trial in acute lymphoblastic leukemia (ALL). The data are described in Chapter 2 of [@BiocCaseStudies]. For the
+
+<figure id="fig:chr12ideogram">
+
+::: center
+```{r chr12ideogram, echo=FALSE, fig.align="center", out.width="100%", out.height="100%"}
+knitr::include_graphics("figures/chr12ideogram.png")
+```
+:::
+
+<figcaption>Ideogram for human chromosome 12. The p arm is on the left, the q arm is on the right, and the centromere is indicated by a notch. The shaded bands together represent 12q21. This band is composed of three sub-bands: 12q21.1, 12q21.2, and 12q21.3. The last of these is composed of sub-sub-bands 12q21.31, 12q21.32, and 12q21.33.</figcaption>
+
+</figure>
````
Replace 56-66 with the code block to generate the figure. Create a code block containing the code in https://github.com/Bioconductor/Category/pull/4/files#diff-fe345a9843884e8331a5988cdd3d46c9275d714c6e87261a3822aa3f63aba921L122-L136. Keep `chr12ideogram` as the name. To display the code properly, you'll have to provide options: `fig.width`, `fig.height`, `fig.align`, and set the text in line 64 as `fig.cap`. Set `echo = FALSE` to prevent display of the code.

kemicky replied:
Ok, thank you very much for your wise guided instructions, I am on it and fixing them immediately . I am grateful.

jwokaty reviewed (changes requested):
I've left comments to help guide you to fix the first figure in the .Rmd file. Part of the problem is that the code is just text. You need to convert code sections into code blocks so that they can be run/compiled. Code in vignettes gradually build up because vignettes explain how to use the functions in the package. If you miss part of the code, the rest of the code below it may not run properly.

You'll need to convert a lot of sections to code blocks. I recommend using the PDF and .Rnw file as guides. Please try these first steps and as I mentioned previously, remove other files. If you are able to get the first figure generating, I recommend trying the same approach with the others.
